app/facades/websocket/WebSocketController.scala

```diff
@@ -174,32 +175,45 @@ class WebSocketController @Inject() (actorSystem:  ActorSystem,
                            overflow:    OverflowStrategy) = {
     import akka.actor.Status._
 
-    val (toSocket, publisher) = Source.actorRef[DSAMessage](bufferSize, overflow)
-      .toMat(Sink.asPublisher(false))(Keep.both).run()(materializer)
+    var (toSocket, publisher) = Source.actorRef[DSAMessage](bufferSize, overflow)
+      .toMat(Sink.asPublisher(false))(Keep.both)
+      .run()(materializer)
+
+    val sink = Sink.actorRef(toSocket, Success(()))
 
     val fRoutee = (registry ? GetOrCreateDSLink(sessionInfo.ci.linkName)).mapTo[Routee]
 
-    fRoutee map { routee =>
-      val wsProps = WebSocketActor.props(toSocket, routee, eventDaos,
-        WebSocketActorConfig(sessionInfo.ci, sessionInfo.sessionId, Settings.Salt))
+    fRoutee flatMap   { routee =>
+
+      val subscriptions = (routee ? GetSubscriptionSource).mapTo[Future[SubscriptionSourceMessage]].flatten
```

Comment:
I'm still not happy that implementation details leak into the controller... Sounds like we can get away with it now, but I'm concerned it will become a problem later.

Reply:
Fair point. To be honest i don't know how to do it better now. Marked it with TODO, marked it in mind as my personal technical debt. Going to keep it as is in this PR if you don't mind.